main/src/ca/uwaterloo/flix/language/errors/ExhaustiveMatchError.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Jason Mitterteiner
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.errors
+
+import ca.uwaterloo.flix.language.CompilationError
+import ca.uwaterloo.flix.language.ast.SourceInput
+import ca.uwaterloo.flix.util.vt.VirtualString._
+import ca.uwaterloo.flix.util.vt.VirtualTerminal
+
+/**
+  * An error raised to indicate a parse error.
+  *
+  * @param msg the error message.
+  * @param src the source input.
+  */
+case class ExhaustiveMatchError(msg: String, src: SourceInput) extends CompilationError {
+  val kind = "Exhaustive Match Error"
+  val source: SourceInput = src
+  val message: VirtualTerminal = {
+    val vt = new VirtualTerminal
+    vt << Line(kind, source.format) << NewLine
+    vt << ">> Exhaustive Match Error:" << NewLine
+    vt << NewLine
+    vt << Red(msg) << NewLine
```

Comment:
What is the "format" of msg? I was thinking that it would be a *list* of patterns or at least a list of strings, e.g.

Non-Exhaustive Match Error:

Unmatched:
  - Foo
  - Bar
  - Baz(_)

by the way Exhaustive Match Error should probably be Non-Exhaustive Match Error?